orthos2/api/tests/test_forms.py

```diff
@@ -0,0 +1,185 @@
+"""Test module to ensure the functionality of the API forms."""
+
+from django.test import TestCase, override_settings
+
+from orthos2.api.forms import *
+
+
+class ReserveMachineAPIFormTests(TestCase):
+    def test_form(self):
+        """Test the machine reservation API form"""
+        # Arrange & Act
+        form = ReserveMachineAPIForm({"reason": "my reason", "until": "9999-12-31", "user": "testuser"})
+
+        # Assert
+        self.assertTrue(form.is_valid())
+
+
+class VirtualMachineAPIFormTests(TestCase):
+    fixtures = [
+        "orthos2/data/fixtures/vendors.json",
+        "orthos2/data/fixtures/tests/test_machines.json",
+        "orthos2/data/fixtures/systems.json",
+    ]
+
+    def test_form(self):
+        """Test the virtual machine creation API form"""
+        # Arrange & Act
+        host = Machine.objects.get_by_natural_key("test.testing.suse.de")
+        form = VirtualMachineAPIForm(
+            {
+                "architecture": "x86_64",
+                "system": "2",
+                "ram_amount": "2048",
+                "disk_size": "30",
+                "image": "none",
+                "networkinterfaces": "2",
+            },
+            **{"virtualization_api": host.virtualization_api}
+        )
+
+        # Assert
+        self.assertTrue(form.is_valid())
+
+
+class MachineAPIFormTests(TestCase):
+    fixtures = [
+        "orthos2/data/fixtures/architectures.json",
+        "orthos2/data/fixtures/systems.json"
+    ]
+
+    def test_form(self):
+        """Test the machine creation API form"""
+        # Arrange & Act
+        form = MachineAPIForm({
+            "fqdn": "test.foo.de",
+            "enclosure": "",
+            "architecture_id": "1",
+            "system_id": "1",
+            "group_id": "none",
+            "check_connectivity": "3"
+        })
+        # This is removing the "validate_dns" validator, mocking this is not
+        # possible since the validator is part of the class definition.
+        form.fields["fqdn"].validators.pop(0)
+
+        # Assert
+        self.assertTrue(form.is_valid())
+
+
+class DeleteMachineAPIFormTests(TestCase):
+
+    fixtures = [
+        "orthos2/data/fixtures/systems.json",
+        "orthos2/data/fixtures/vendors.json",
+        "orthos2/data/fixtures/tests/test_machines.json"
+    ]
+
+    def test_form(self):
+        """test the machine deletion API form"""
+        # Arrange & Act
+        form = DeleteMachineAPIForm({"fqdn": "test.testing.suse.de"})
+
+        # Assert
+        self.assertTrue(form.is_valid())
+
+
+class SerialConsoleAPIFormTests(TestCase):
+    fixtures = ["orthos2/data/fixtures/serialconsoletypes.json"]
+
+    def test_form(self):
+        """Test the serial console creation API form"""
+        # Arrange & Act
+        form = SerialConsoleAPIForm({
+            "stype": "1",
+            "baud_rate": "57600",
+            "kernel_device": "ttyS",
+            "kernel_device_num": "5"
```

Comment:
I would use a valid kernel_device_num for ttySX.
x86 typically have ttyS0 and ttyS1.

Reply:
While you are correct that generally 0 or 1 is used, 5 is fully valid tty as well.